src/Microsoft.AspNetCore.TestHost/WebSocketClient.cs

```diff
@@ -181,12 +190,20 @@ bool IHttpWebSocketFeature.IsWebSocketRequest
 
             Task<WebSocket> IHttpWebSocketFeature.AcceptAsync(WebSocketAcceptContext context)
             {
-                Context.HttpContext.Response.StatusCode = 101; // Switching Protocols
-
                 var websockets = TestWebSocket.CreatePair(context.SubProtocol);
-                _clientWebSocketTcs.SetResult(websockets.Item1);
-                _serverWebSocket = websockets.Item2;
-                return Task.FromResult<WebSocket>(_serverWebSocket);
+                if (_clientWebSocketTcs.TrySetResult(websockets.Item1))
+                {
+                    Context.HttpContext.Response.StatusCode = 101; // Switching Protocols
+                    _serverWebSocket = websockets.Item2;
+                    return Task.FromResult<WebSocket>(_serverWebSocket);
+                }
+                else
+                {
+                    Context.HttpContext.Response.StatusCode = 500; // Internal Server Error
+                    websockets.Item1.Dispose();
+                    websockets.Item2.Dispose();
+                    return _clientWebSocketTcs.Task; // Cancelled or Faulted - no result
```

Comment:
nit: we prefer using the American English spelling of words so Cancelled => Canceled.